### PR TITLE
Connect products to store

### PR DESCRIPTION
**Pull request by DruDro**

### CODE REVIEW

File: src/routes/index.tsx [hunk 1]
```diff
@@ -8,10 +8,10 @@ import About from './about';
 export default () => {
 	return (
 		<section>
-			<Route exact={true} path="/" component={Home} />
-			<Route path="/about" component={About} />
-			<Route exact={true} path="/products" component={Products} />
-			<Route path="/products/:id" component={Product}  />
+			<Route exact={true} path="/" component={() => <Home />} />
+			<Route path="/about" component={() => <About />} />
+			<Route exact={true} path="/products" component={() => <Products />} />
```
[the remark this answers is not in the document]
Comment (DruDro) [reply src/routes/index.tsx]: I get this warning if I pass just a component:
index.js:1437 Warning: Failed prop type: Invalid prop `component` of type `object` supplied to `Route`, expected `function`.

[the remark this answers is not in the document]
Comment (iliatcymbal) [reply src/routes/index.tsx]: Try to update `react-router-dom` to 5.0.1 version

File: src/store/products/actions.ts [hunk 2]
```diff
@@ -12,3 +13,8 @@ export const productsSuccessAction = (data: IProduct[]) => ({
 	type: SUCCESS,
 	data,
 });
+
+export const clearProductsAction = (data: []) => ({
```
Comment (iliatcymbal) [inline src/store/products/actions.ts]: no need any parameters in clear method

File: src/store/products/reducers.ts [hunk 3]
```diff
@@ -39,12 +39,23 @@ export function productsReducers(state: State = initialState, action: IAction) {
...
+		case CLEAR:
+			return {
+				...state,
+				data: action.data,
+				status: {
+					...state.status,
+					loading: false,
+					loaded: false,
+				},
+			};
```
Comment (iliatcymbal) [inline src/store/products/reducers.ts]: it should be 
```
{
  ...state,
  ...initialState,
}
```

or even just
```
{
  ...initialState,
}
```

File: src/routes/products/Component.tsx [hunk 4]
```diff
@@ -6,25 +6,31 @@ import { TComponentProps } from './type';
...
+const ProductList: React.FunctionComponent<TComponentProps> = ({ products, status, getProducts, clearProducts }) => {
 	useEffect(() => {
-		if (!products.length) {
-			getProducts();
-		}
+		getProducts();
 		// eslint-disable-next-line
 	}, []);
+	// eslint-disable-next-line
+	useEffect(() => () => { clearProducts(); }, []);
```
Comment (iliatcymbal) [inline src/routes/products/Component.tsx]: You have  another  useEffect. Why need the second?

File: src/routes/products/Component.tsx [hunk 5]
```diff
@@ -6,25 +6,31 @@ import { TComponentProps } from './type';
...
-							<ProductItem id={item.id} kind={item.kind} area-house={item.specification.area} />
-						</li>
-					) : '...loading'
+					status.loading ? '...Loading' :
+						products.length ? products.map(item =>
+							<li key={item.id} className="product-list__product">
+								<ProductItem
+									id={item.id}
+									kind={item.kind}
+									area-house={item.specification.area}
```
Comment (iliatcymbal) [inline src/routes/products/Component.tsx]: what's for this attribute?

Comment (DruDro) [reply src/routes/products/Component.tsx]: I was trying to figure out how to pass this and other data to product items in the list

**Pull request merged**